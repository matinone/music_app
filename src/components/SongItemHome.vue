<template>
  <li
    class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50 example"
    :id="`song-id-${song.docId}`"
    :class="`song-id-${song.docId}`"
  >
    <div>
      <router-link
        :to="{ name: 'song', params: { id: song.docId } }"
        class="font-bold block text-gray-600"
      >
        {{ song.modifiedName }}
      </router-link>
      <span class="text-gray-500 text-sm italic">
        Uploaded by {{ song.userDisplayName }}
      </span>
    </div>

    <div class="text-gray-600 text-lg">
      <span class="comments">
        <i class="fa fa-comments text-gray-600"></i>
        {{ song.commentCount }}
      </span>
    </div>
  </li>
</template>

<script>
export default {
  name: "SongItemHome",
  props: {
    song: {
      type: Object,
      required: true,
    },
  },
};
</script>
