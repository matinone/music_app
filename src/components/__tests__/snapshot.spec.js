import { it, expect, describe } from "vitest";
import { shallowMount, RouterLinkStub } from "@vue/test-utils";

import SongItemHome from "@/components/SongItemHome.vue";

describe("Snapshots SongItemHome.vue", () => {
  it("Render correctly", () => {
    const song = {
      docId: "abc123",
      modifiedName: "test",
      userDisplayName: "test",
      commentCount: 4,
    };

    const wrapper = shallowMount(SongItemHome, {
      props: { song },
      global: {
        components: {
          "router-link": RouterLinkStub,
        },
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
