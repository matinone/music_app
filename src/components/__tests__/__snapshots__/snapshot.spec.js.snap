// Vitest Snapshot v1

exports[`Snapshots SongItemHome.vue > Render correctly 1`] = `
<li
  class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50 example song-id-abc123"
  id="song-id-abc123"
>
  <div>
    <router-link-stub
      class="font-bold block text-gray-600"
      custom="false"
      to="[object Object]"
    />
    <span
      class="text-gray-500 text-sm italic"
    >
       Uploaded by test
    </span>
  </div>
  <div
    class="text-gray-600 text-lg"
  >
    <span
      class="comments"
    >
      <i
        class="fa fa-comments text-gray-600"
      />
       4
    </span>
  </div>
</li>
`;
